### ISSUE TITLE
Deprecate raw `RuntimeDbWeight::reads_writes` calls

### ISSUE THREAD

**Issue by ggwpez** [paritytech/polkadot-sdk#378]
We often use these functions in migrations to avoid writing benchmarks. This worked mostly fine with 1D weights but does not work for chromatic weights.  

The public `reads_writes, reads, writes` functions should be deprecated and the weight templates need to find another way of accessing these constants.

**Comment by ybensacq** [comment]
@ggwpez the purpose of this task is just to deprecate the 3 functions ?

**Comment by ggwpez** [comment]
Not quite... we also need to come up with an alternative for weight templates since they have a legitimate use-case to access these constants. I dont have a good idea, but maybe you can start by just removing this thing?  https://github.com/paritytech/polkadot-sdk/blob/552be4800d9e4b480f79a300fc531783e04be099/substrate/primitives/weights/src/lib.rs#L75  

If the `contracts` pallet depends on that then we copy it into it, but otherwise i hope we dont use it anywhere.

**Comment by kianenigma** [comment]
https://github.com/paritytech/polkadot-sdk/pull/3640 will partually address this.
